### PR TITLE
Fix to #8186 - Navs: nav expansion on GJ QSRE in projection doesn't introduce parameter

### PR DESCRIPTION
**Pull request by maumar**
Problem was that when trying to find query to bind with via outer parameter we were too strict and would not find the appropriate query. However it is also safe to bind to grouping qsre, even though it is wrapped by client GroupJoin.

This also enables additional include scenarios when GJ qsre is projected - fixed a small bug in this area that was exposed by the main fix.

### CODE REVIEW

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 1]
```diff
@@ -1969,19 +1969,20 @@ public override Expression BindMemberToValueBuffer(MemberExpression memberExpres
 
         private ParameterExpression BindPropertyToOuterParameter(IQuerySource querySource, IProperty property, bool isMemberExpression)
         {
-            if (querySource != null && _canBindPropertyToOuterParameter)
+            if (querySource != null && _canBindPropertyToOuterParameter && ParentQueryModelVisitor != null)
```
Comment (maumar) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: filed #8790 to track enabling this for broader set of scenarios - too risky/too much work to do it now.

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 2]
```diff
@@ -2020,6 +2021,20 @@ var parameterWithSamePrefixCount
             return null;
         }
 
+        private bool CanBindToQuerySourceUsingOuterParam(IQuerySource querySource)
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: `CanBindToParentQMV`

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 3]
```diff
@@ -1969,19 +1969,20 @@ public override Expression BindMemberToValueBuffer(MemberExpression memberExpres
 
         private ParameterExpression BindPropertyToOuterParameter(IQuerySource querySource, IProperty property, bool isMemberExpression)
         {
-            if (querySource != null && _canBindPropertyToOuterParameter)
+            if (querySource != null && _canBindPropertyToOuterParameter && ParentQueryModelVisitor != null)
             {
-                var outerQueryModelVisitor = ParentQueryModelVisitor;
-                var outerSelectExpression = outerQueryModelVisitor?.TryGetQuery(querySource);
+                var isBindable = CanBindToQuerySourceUsingOuterParam(querySource);
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: variable/parameters not prefixed?

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs (hunk 2 above)
Comment (maumar) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: we already have property CanBindToParentQueryModel that is being used for non outer-param binding. How about CanBindToParentQueryModelUsingOuterParam?

Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: but but we need `parameter` full not `param` 😢

Comment (maumar) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: yup, used "parameter" in the final name

**Pull request merged**